### PR TITLE
specifications: default tracks

### PR DESCRIPTION
**Pull request by sergiusens**
Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

File: specifications/default-tracks.org [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+#+END_SRC
+
+When setting a valid track, this is what the command shall output:
+
+#+BEGIN_SRC sh
+$ snapcraft set-default-track node 17
+Default track for node has been set to 17
+#+END_SRC
+
+If the track is no valid, the output shall follow the convention following by
```
Comment (cjp256) [inline specifications/default-tracks.org:55]:
```suggestion
If the track is not valid, the output shall follow the convention following by
```

File: specifications/default-tracks.org [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  14       hidden
+  13       closed
+#+END_SRC
+
+#+BEGIN_QUOTE
+The =hidden= and =closed= Status values are shown for illustration purposes, only
+active and default shall be implemented as part of this specification.
+#+END_QUOTE
+
+** Setting a default track
```
Comment (tbille) [inline specifications/default-tracks.org:41]: is it worth mentioning how to "remove " a default track by pointing it to latest?

Comment (sergiusens) [reply specifications/default-tracks.org:41]: I thought it was agreed that this was not a "removal", please correct me if I am wrong.

Comment (tbille) [reply specifications/default-tracks.org:41]: no ideed it's not a removal.

I was just wondering if you were capturing also those details of the spec in here: ~"To revert back your change to the behaviour, the user has to set the default track to latest." 

So basically there is no removal, just setting the default track back the latest branch.

File: specifications/default-tracks.org [hunk 3]
```diff
@@ -73,15 +73,15 @@ Snap Store APIs already exist to implement /setting/ a default track to a given
...
-
 Displaying of track information shall be resolved by the ~channel-map~ endpoint
 defined in the V2 Snap Store API [[https://dashboard.snapcraft.io/docs/v2/en/snaps.html#snap-channel-map][Snap Channel Map]]. For this first implementation
 the default track shall be retrieved from ~snap.default-track~, leaving room for a
 future change to account for ~hidden~ and ~closed~ channel attributes.
 
+In the future, the [[https://dashboard.snapcraft.io/docs/v2/en/snaps.html#snap-channel-map][Snap Channel Map]] endpoint shall add a ~status~ attribute for
+tracks which shall return: =active=, =hidden= and =closed=. In this current
+implementation, the Snapcraft logic shall query for this track attribute and
+return a default of =active= for when the attribute is not found.
```
Comment (cjp256) [inline specifications/default-tracks.org]:
```suggestion
return a default of =active= when the attribute is not found.
```

File: specifications/default-tracks.org [hunk 4]
```diff
@@ -74,14 +74,15 @@ track. This is provided by the ~/metadata~ endpoint documented as part of the V1
...
-tracks which shall return: =active=, =hidden= and =closed=. In this current
-implementation, the Snapcraft logic shall query for this track attribute and
-return a default of =active= for when the attribute is not found.
+defined in the V2 Snap Store API [[https://dashboard.snapcraft.io/docs/v2/en/snaps.html#snap-channel-map][Snap Channel Map]].
+
+For this first implementation the default track shall be retrieved from the
+=snap.default-track= field. In the future, the [[https://dashboard.snapcraft.io/docs/v2/en/snaps.html#snap-channel-map][Snap Channel Map]] endpoint shall add
+a =status= field for a track list item which shall return: =active=, =hidden= and
+=closed=. In this current implementation, the Snapcraft logic shall query for this
+field and set a fallback value of =active= for when the =status= field is not found.
```
Comment (cjp256) [inline specifications/default-tracks.org:83]:
```suggestion
field and set a fallback value of =active= when the =status= field is not found.
```

File: specifications/default-tracks.org [hunk 5]
```diff
@@ -74,14 +74,15 @@ track. This is provided by the ~/metadata~ endpoint documented as part of the V1
...
+The =Created= information shall be retrieved from the =creationd-date= field for a track
```
Comment (cjp256) [inline specifications/default-tracks.org:84]:
```suggestion
The =Created= information shall be retrieved from the =creation-date= field for a track
```

**Pull request merged**